controllers/tests/cruisecontroloperation_ttl_controller_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright © 2022 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/banzaicloud/koperator/api/v1alpha1"
+	"github.com/banzaicloud/koperator/api/v1beta1"
+	"github.com/banzaicloud/koperator/controllers"
+	"github.com/banzaicloud/koperator/pkg/util"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+var _ = Describe("CruiseControlTaskReconciler", func() {
+	var (
+		count              uint64 = 0
+		namespace          string
+		namespaceObj       *corev1.Namespace
+		kafkaClusterCRName string
+		kafkaCluster       *v1beta1.KafkaCluster
+		opName             string = "operation1"
+	)
+	BeforeEach(func() {
+		atomic.AddUint64(&count, 1)
+
+		namespace = fmt.Sprintf("cruisecontroloperationttl-%v", count)
+		namespaceObj = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+		kafkaClusterCRName = controllers.CruiseControlTaskTestKafkaClusterName
+		kafkaCluster = createMinimalKafkaClusterCR(kafkaClusterCRName, namespace)
+	})
+
+	JustBeforeEach(func() {
+		By("creating namespace " + namespace)
+		err := k8sClient.Create(context.TODO(), namespaceObj)
```

Comment:
**Recommended**: We also **SHOULD** use [`context.Background()`](https://pkg.go.dev/context#Background) in these cases unless we intend to change these in the foreseeable future, because we don't expect to time out in these tests.

> Background returns a non-nil, empty Context. It is never canceled, has no values, and has no deadline. It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.

The same holds true for the reset of the cases.

Reply:
context.Background() and context.TODO() is completely same.
https://go.dev/src/context/context.go  line 199